File: Integraupt-Backend/BackendAdReserva/src/main/java/com/AdReserva/model/Facultad.java
package com.AdReserva.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "facultad")
public class Facultad {

    @Id
    @Column(name = "IdFacultad")
    private Integer id;

    @Column(name = "Nombre", nullable = false)
    private String nombre;

    @Column(name = "Abreviatura")
    private String abreviatura;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getAbreviatura() {
        return abreviatura;
    }

    public void setAbreviatura(String abreviatura) {
        this.abreviatura = abreviatura;
    }
}

File: Integraupt-Backend/servicio-olimpiadas/app/Models/Facultad.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Facultad extends Model
{
    protected $table = 'facultad';

    protected $primaryKey = 'IdFacultad';

    public $timestamps = false;

    protected $fillable = [
        'Nombre',
        'Abreviatura',
    ];
}
